operation.go

```diff
@@ -35,12 +38,44 @@ func update(ctx context.Context, c *Collection, model Model, opts ...*options.Up
 		return err
 	}
 
-	res, err := c.UpdateOne(ctx, bson.M{field.ID: model.GetID()}, bson.M{"$set": model}, opts...)
+	query := bson.M{field.ID: model.GetID()}
+	modelVersionable, isVersionable := model.(Versionable)
```

Comment:
What do you think about a little bit of change in variable names,...? (and maybe we should convert it to a method like `query,isVersionable:=updateVersion(m)`)

```golang
var v interface{}
vmodel, isVersionable := model.(Versionable)
if isVersionable{
v = modelVersionable.GetVersion()
if vmodel.IsVersionZero() {
    query["$or"] = bson.A{
    bson.M{vmodel.GetVersionFieldName(): v},
    bson.M{vmodel.GetVersionFieldName(): bson.M{"$exists": false}},
   }
} else {
	query[vmodel.GetVersionFieldName()] = v
}

vmodel.IncrementVersion()
}
```

Reply:
No problem for variable names.
Not sure the additional method is really worth it, I personnally find it clearer without as its only meant to be used here at this stage.